include/alp/reference/matrix.hpp

```diff
@@ -2444,6 +2444,91 @@ namespace alp {
 
 		}
 
+		/**
+		 * Calculates the iteration space for row-dimension for the given matrix and band index.
+		 *
+		 * @tparam MatrixType The type of ALP matrix
+		 * @tparam band_index The index of the desired matrix band
+		 *
+		 * @param[in] A       ALP matrix
+		 *
+		 * @returns a pair of size_t values,
+		 *          the first representing lower and the second upper limit.
+		 */
+		template<
+			size_t band_index, typename MatrixType,
+			std::enable_if_t<
+				is_matrix< MatrixType >::value
+			> * = nullptr
+		>
+		std::pair< size_t, size_t > calculate_row_coordinate_limits( const MatrixType &A ) {
+
+			using Structure = typename MatrixType::structure;
+
+			static_assert(
+				band_index < std::tuple_size< typename Structure::band_intervals >::value,
+				"Provided band index is out of bounds."
+			);
+
+			const std::ptrdiff_t l = structures::get_lower_bandwidth< band_index >( A );
```

Comment:
`get_lower_bandwidth` name is a bit confusing, perhaps it would be good to change this name (and `get_upper_bandwidth` )